### PR TITLE
Use operation: key_add for cuda

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
Ahhh great! I had this change on my machine locally but never pushed it. :)

**Comment by beckermr** [comment]
I could not find my local edits that had this but LGTM!

**Review by beckermr — APPROVED** [review]

File: recipe/migrations/cuda111_112.yaml [hunk 1]
```diff
@@ -8,93 +8,21 @@ __migrator:
...
-      - quay.io/condaforge/linux-anvil-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cos7-cuda:10.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cos7-cuda:10.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cos7-cuda:10.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cuda:11.0    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cuda:11.1    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-      - quay.io/condaforge/linux-anvil-cuda:11.2    # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64"]
-    cuda_compiler_version:
```
Comment (beckermr) [inline recipe/migrations/cuda111_112.yaml:37]: I think we may still want the cuda compiler version block for ordering. Might be good to test if you are not sure.

Comment (isuruf) [reply recipe/migrations/cuda111_112.yaml:37]: I checked and it's not

Comment (beckermr) [reply recipe/migrations/cuda111_112.yaml:37]: SGTM!

**Comment by jakirkham** [comment]
That's much cleaner! Thanks for working on this Isuru 😄

**Comment by isuruf** [comment]
Merge?

**Comment by beckermr** [comment]
Sure yeah. I forgot about the primary key field.

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
Not sure if this is related, but am seeing CentOS 6 pop-up in CUDA 11.0 builds after re-rendering ( https://github.com/conda-forge/ucx-split-feedstock/pull/87 ). Didn't happen for other CUDA versions. So wondering if one of the list orderings is off

**Comment by isuruf** [comment]
It's not. Need https://github.com/conda-forge/ucx-split-feedstock/pull/87/commits/63c828e3c4750c1ac77582c549ef15254493f5cf applied here.

**Comment by jakirkham** [comment]
Ah ok. Thanks for figuring that out 😄

I tried to copy the latest migrators over, but was still running into issues. Wondering if we should transfer those changes over here

**Comment by jakirkham** [comment]
Capturing those changes in PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1368 )

**Pull request merged**